### PR TITLE
LC-532: fix leak of threads in TextExtractor stage

### PR DESCRIPTION
**Pull request by Jozurf**
- closing of inputStream is in a separate branch LC-528. This branch changes how it got inputStream from filepath
- changed test for TextExtractor to use absolute path rather than relative path to reflect the same procedure. (doc.getString(filePathField) would return an absolute path)

### CODE REVIEW

File: lucille-plugins/lucille-tika/src/test/java/com/kmwllc/lucille/tika/stage/TextExtractorTest.java
Comment (rseitz) [inline lucille-plugins/lucille-tika/src/test/java/com/kmwllc/lucille/tika/stage/TextExtractorTest.java:115]: try creating a Path from the relative path string ("src/test/...") and then calling toAbsolutePath() on it -- can do this in one line

Comment (Jozurf) [reply lucille-plugins/lucille-tika/src/test/java/com/kmwllc/lucille/tika/stage/TextExtractorTest.java]: resolved in new commit

File: lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java
Comment (rseitz) [inline lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java:138]: The original FileUtils.getInputStream had extra functionality that we're not yet capturing here. If the filePath wasn't a valid URI it would treat it as a regular file path and open a FileInputStream...

File: lucille-core/src/main/java/com/kmwllc/lucille/util/FileUtils.java
Comment (rseitz) [inline lucille-core/src/main/java/com/kmwllc/lucille/util/FileUtils.java:71]: add new parameter

File: lucille-core/src/main/java/com/kmwllc/lucille/util/FileUtils.java
Comment (rseitz) [inline lucille-core/src/main/java/com/kmwllc/lucille/util/FileUtils.java:97]: consider removing this inner try/catch since the outer try/catch is for the same exception type

File: lucille-core/src/main/java/com/kmwllc/lucille/util/FileUtils.java
Comment (rseitz) [inline lucille-core/src/main/java/com/kmwllc/lucille/util/FileUtils.java:88]: is this existence test necessary, or if the file doesn't exist and we call file.getContent() will we get a FileSystemException that will be caught and logged

Comment (rseitz) [reply lucille-core/src/main/java/com/kmwllc/lucille/util/FileUtils.java:88]: looking at docs for the individual methods we're calling here, can we simplify the code as follows?

`      try {
        file = fsManager.resolveFile(path); // never returns null
        FileContent content = file.getContent(); // can be called even if !file.exists()
        return content.getInputStream(); // should throw FileSystemException if !file.exists()
      } catch (FileSystemException e) {
        log.warn("Error retrieving file contents: {}", path, e);
        return null;
      }`

File: lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java
Comment (rseitz) [inline lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java:137]: suggest adding comment that file.getContent() never returns null; can include link to vfs doc where it says this

File: lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java
Comment (rseitz) [inline lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java:147]: add comment saying parseInputStream would close the InputStream itself
add comment saying file.close() will close FileContent and InputStream as well?

**Review by rseitz — APPROVED** [review]
approving but please wait for second approval from kyle before merging

File: lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java
Comment (kyle-kmw) [inline lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java:150]: If an exception is thrown. file won't close. Better to wrap the creation of FileObject file in a try-with-resources and then we don't have to call close at all and we know that try will close the resource no matter what happens e.g.

try (FileObject file = FileUtils.getFileObject(filePath, fsManager)) {
 ...
}

File: lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java
Comment (kyle-kmw) [inline lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java:136]: recommend popping the inputstream into the try

try (InputStream inputStream = file.getContent().getInputStream()) {
 ...
}

File: lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java
Comment (kyle-kmw) [reply lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java:150]: I suggested above creating the inputStream in a try-with-resources. The close() contract is a no-op if it is called multiple times. If some day parseInputStream is refactored and the implementation changes so that it doesn't close the inputStream (which would be a reasonable change IMO) then we at least know that the resource we created here is closed before we leave.

**Review by kyle-kmw — APPROVED** [review]

**Pull request merged**
